orchestrator/test_runner/src/ibc_auto_forward.rs

```diff
@@ -0,0 +1,1070 @@
+use crate::airdrop_proposal::wait_for_proposals_to_execute;
+use crate::happy_path::send_erc20_deposit;
+use crate::utils::*;
+use crate::OPERATION_TIMEOUT;
+use crate::{
+    get_ibc_chain_id, one_eth, ADDRESS_PREFIX, COSMOS_NODE_GRPC, IBC_ADDRESS_PREFIX, IBC_NODE_GRPC,
+    STAKING_TOKEN,
+};
+use clarity::Address as EthAddress;
+use cosmos_gravity::utils::decode_any;
+use deep_space::address::Address as CosmosAddress;
+use deep_space::error::CosmosGrpcError;
+use deep_space::private_key::PrivateKey as CosmosPrivateKey;
+use deep_space::utils::encode_any;
+use deep_space::{Coin as DSCoin, Contact, Msg};
+use gravity_proto::cosmos_sdk_proto::bech32ibc::bech32ibc::v1::UpdateHrpIbcChannelProposal;
+use gravity_proto::cosmos_sdk_proto::cosmos::bank::{
+    v1beta1 as Bank, v1beta1::query_client::QueryClient as BankQueryClient,
+};
+use gravity_proto::cosmos_sdk_proto::cosmos::base::v1beta1::Coin;
+use gravity_proto::cosmos_sdk_proto::ibc::applications::transfer::v1::MsgTransfer;
+use gravity_proto::cosmos_sdk_proto::ibc::applications::transfer::{
+    v1 as IbcTransferV1, v1::query_client::QueryClient as IbcTransferQueryClient,
+};
+use gravity_proto::cosmos_sdk_proto::ibc::core::channel::v1::query_client::QueryClient as IbcChannelQueryClient;
+use gravity_proto::cosmos_sdk_proto::ibc::core::channel::v1::{
+    QueryChannelClientStateRequest, QueryChannelsRequest,
+};
+use gravity_proto::cosmos_sdk_proto::ibc::lightclients::tendermint::v1::ClientState;
+use gravity_proto::gravity::query_client::QueryClient as GravityQueryClient;
+use gravity_proto::gravity::{
+    MsgExecuteIbcAutoForwards, PendingIbcAutoForward, QueryPendingIbcAutoForwards,
+};
+use gravity_utils::error::GravityError;
+use gravity_utils::num_conversion::one_atom;
+use num256::Uint256;
+use std::ops::{Add, Mul};
+use std::str::FromStr;
+use std::time::Instant;
+use std::time::{Duration, SystemTime};
+use tokio::time::sleep as delay_for;
+use tonic::transport::Channel;
+use web30::client::Web3;
+
+pub const MSG_TRANSFER_TYPE_URL: &str = "/ibc.applications.transfer.v1.MsgTransfer";
+pub const MSG_EXECUTE_IBC_AUTO_FORWARDS_TYPE_URL: &str = "/gravity.v1.MsgExecuteIbcAutoForwards";
+pub const UPDATE_HRP_IBC_CHANNEL_PROPOSAL: &str =
```

Comment:
ideally we would create utility functions in the appropraite deep-space / cosmos-gravity files for these.